src/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1940,6 +1946,8 @@ private function getArrayBindingType(ParameterType|int|string $type): ArrayParam
             ParameterType::STRING => ArrayParameterType::STRING,
             ParameterType::INTEGER => ArrayParameterType::INTEGER,
             ParameterType::ASCII => ArrayParameterType::ASCII,
+            ParameterType::BINARY => ArrayParameterType::BINARY,
+            default => throw new QueryException('Unsupported type for array parameter'),
```

Comment:
Why do we need the `default` branch here?

Reply:
The default is needed to handle $type values that do not match any of the match cases. It ensures reliability, allows throwing clear exceptions for unsupported types, and prevents errors if the enum is modified or unexpected values are introduced.